nuxeo-services/nuxeo-platform-web-common/src/main/java/org/nuxeo/ecm/platform/web/common/requestcontroller/service/NuxeoCorsFilterDescriptor.java

```diff
@@ -74,15 +79,38 @@
     @XNode("@maxAge")
     protected int maxAge = -1;
 
-    protected String pattern = "";
+    protected Pattern pattern;
 
     @XNode("pattern")
-    public void setPattern(String pattern) {
-        this.pattern = Framework.expandVars(pattern);
+    public void setPattern(String patternString) {
+        patternString = Framework.expandVars(patternString);
+        if (!StringUtils.isBlank(patternString)) {
+            pattern = Pattern.compile(patternString);
+        }
+    }
+
+    // volatile for double-checked locking
+    protected volatile CORSFilter filter;
+
+    public CORSFilter getFilter() {
+        if (filter == null) {
+            synchronized (this) {
```

Comment:
```
private Object lock = new Object();

public CORSFilter getFilter() {
  // ...
  synchronized(lock) {
  }
  // ...
}

```

Reply:
What you're suggesting is different, it locks on all access. This is expensive. The code I have uses the so-called "double-checked locking" idiom which is faster and correct, provided the field used is `volatile` and there's a single assignment to it and it's not mutated after being assigned.

Reply:
@efge , i'm just saying to replace the `synchronized(this)` with an internal Lock, that's thy i put the `/...`

Reply:
Having an extra lock object is not needed.